classes.py

```diff
@@ -28,9 +34,28 @@ class ChangeSet():
     """Pages that were added in the diff"""
     removed: List[Page]
     """Pages that were removed in the diff"""
-    changed: List[Tuple[Page, List[PropertyChange]]]
+    changed: List[Tuple[Page, List[PropertyChange], datetime]]
     """Pages that had their properties changed in the diff. The key is the page id and the value is the list of 
     properties that were changed"""
 
     def is_empty(self) -> bool:
         return len(self.added) == 0 and len(self.removed) == 0 and len(self.changed) == 0
+
+    def merge_set(self, other_changed_set):
+        """merges the changes from the other set into the current set
+        """
+
+        # For the added and removed pages, we simply add them to the new set if there is any
+        self.added += other_changed_set.added
```

Comment:
I feel like this is not necessary since you don't have timeout to deal with for page additions/removals and therefore don't pass them forward to the next changeset

Reply:
We do, because what we intend to do is always forward those into the master changeset, since our current logic is 
get_new_changes -> merge_to_master -> check_timeout_at_master -> send_message_from_master